test/parallel/test-runner-worker-id.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const test = require('node:test');
+const { spawnSync, spawn } = require('child_process');
+const { join } = require('path');
+const fixtures = require('../common/fixtures');
+const testFixtures = fixtures.path('test-runner');
+
+test('testing with isolation enabled', () => {
+  const args = ['--test', '--experimental-test-isolation=process'];
+  const child = spawnSync(process.execPath, args, { cwd: join(testFixtures, 'worker-id') });
+
+  assert.strictEqual(child.stderr.toString(), '');
+  const stdout = child.stdout.toString();
+
+  assert.match(stdout, /NODE_TEST_WORKER_ID 1/);
+  assert.match(stdout, /NODE_TEST_WORKER_ID 2/);
+  assert.match(stdout, /NODE_TEST_WORKER_ID 3/);
+
+  assert.strictEqual(child.status, 0);
+});
+
+test('testing with isolation disabled', () => {
+  const args = ['--test', '--experimental-test-isolation=none'];
+  const child = spawnSync(process.execPath, args, { cwd: join(testFixtures, 'worker-id') });
+
+  assert.strictEqual(child.stderr.toString(), '');
+  const stdout = child.stdout.toString();
+  const regex = /NODE_TEST_WORKER_ID 1/g;
+  const result = stdout.match(regex);
+
+  assert.strictEqual(result.length, 3);
+  assert.strictEqual(child.status, 0);
+});
+
+
+test('testing with isolation enabled in watch mode', async () => {
```

Comment:
The watch mode tests added here have at least two problems:

1. They don't actually test watch mode beyond passing the `--watch` flag. To properly test watch mode, you have to trigger the restart functionality.
2. These tests will almost certainly be flaky in the CI. Waiting for a second, killing the child process, and expecting it to have run successfully probably works locally, but the CI machines can be very slow. I don't think we want to have a timeout at all.

I think these need to be refactored so that:
- The test runs once.
- Once the expected output is seen, a source file is modified (we don't want to modify the fixtures files directly either) to trigger a restart.
- The test runs again. We need the output to be the same both times.